R/tidy.R

```diff
@@ -169,9 +169,55 @@ tidy_block = function(
   if (length(exprs) == 0) return(character(0))
   exprs = if (arrow) replace_assignment(exprs) else as.list(exprs)
   deparse = if (inherits(width, 'AsIs')) deparse2 else base::deparse
+
   unlist(lapply(exprs, function(e) {
     x = deparse(e, width)
     x = reindent_lines(x, indent)
+    x = unlist(lapply(
+      x,
+      function(line){
+
+        is_comment = grepl(mat.comment,
+                           line)
+
+        is_special_comment = grepl("^#+[-'+]",
+                                   trimws(gsub(pat.comment,
+                                               '',
+                                               line)))
+
+        is_regular_comment = is_comment & !is_special_comment
+
+        # ignore code and special comment that starts with #'
+        if (is_regular_comment) {
+
+          # strip away masks
+          line = gsub(pat.comment, '', line)
+
+          # extract indent & comment characters
+          indent_and_comment_characters = sub('(#+).*', '\\1', line)
+
+          # remove indent & comment characters
+          line = sub(indent_and_comment_characters, '', line)
+
+          # wrap line
+          line = unlist(strwrap(line,
```

Comment:
maybe it should be `simplify = TRUE` with no `unlist()` but I was afraid I might miss something here.

Reply:
Try to go without `unlist()` + `simplify = FALSE`. I guess it should be fine.